src/SourceFile.js

```diff
@@ -0,0 +1,229 @@
+/*
+ * SourceFile.js - Represent an ilib-lint rule
+ *
+ * Copyright © 2024 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import NotImplementedError from "./NotImplementedError.js";
+
+/**
+ * @class Represent a source file that can be linted in some way.
+ */
+class SourceFile {
+    /**
+     * Construct a new source file instance. Parsers should produce these.
+     *
+     * @param {String} uri URI or path to the source file
```

Comment:
are we actually going to support URIs?

Reply:
Not in this class, but I can see subclasses in the future that do things like load rows from a database table and store them internally as a javascript object.